src/services/music.player.service.ts

```diff
@@ -0,0 +1,176 @@
+  ChatInputCommandInteraction,
+  GuildMember,
+} from 'discord.js';
+import { LoadType, Shoukaku, Track } from 'shoukaku';
+import { songInfoEmbed } from 'utils/embeds';
+import logger from 'utils/logger';
+
+export const play = async (
+  interaction: ChatInputCommandInteraction,
+  searchPhrase: string,
+  shoukaku: Shoukaku,
+) => {
+  const guildMember = interaction.member as GuildMember;
+
+  if (!guildMember?.voice?.channel?.id) {
+    await interaction.reply({
+      content: `You're not connected to any voice channel! >:(`,
+      ephemeral: true,
+    });
+
+    return;
+  }
+
+  await shoukaku.leaveVoiceChannel(interaction.guildId as string);
+
+  const player = await shoukaku.joinVoiceChannel({
+    guildId: interaction.guildId as string,
+    channelId: guildMember.voice?.channel?.id as string,
+    shardId: 0,
+  });
+
+  const node = shoukaku.options.nodeResolver(shoukaku.nodes);
+
+  /**
+   * Deferring reply, as interaction only waits 3s
+   * for a reply before it fails. Doing so because
+   * song searching can take some time.
+   */
+  await interaction.deferReply();
+
+  /**
+   * If it's a URL, pass it to Lavalink to resolve directly
+   * Otherwise, try youtube music, otherwise last resort -> youtube lyrics
+   */
+  if (!isURL(searchPhrase)) searchPhrase = 'ytmsearch: ' + searchPhrase;
+  let result = await node?.rest.resolve(searchPhrase);
+
+  if (result?.loadType !== LoadType.SEARCH && result?.loadType !== LoadType.TRACK) {
+    searchPhrase = searchPhrase.replace('ytmsearch', 'ytsearch') + ' lyrics';
+    result = await node?.rest.resolve(searchPhrase);
+  }
+
+  let track: Track;
+
+  if (result?.loadType === LoadType.SEARCH) {
+    // if it's multiple search results, take first one
+    track = result.data.shift() as Track;
+  } else if (result?.loadType === LoadType.TRACK) {
+    track = result.data as Track;
+  } else if (
+    result?.loadType === LoadType.PLAYLIST ||
+    result?.loadType === LoadType.EMPTY
+  ) {
+    await interaction.editReply({
+      content: `Didn't really find the song you're looking for :(`,
+    });
+
+    return;
+  } else {
+    throw new Error('An error occurred while trying to play that song');
+  }
+
+  await Promise.all([
+    interaction.editReply({
+      embeds: [decorateEmbed(songInfoEmbed, track)],
+    }),
+    player.playTrack({ track: { encoded: track.encoded } }),
+  ]);
+
+  // TODO: Implement pause/play/stop buttons
+};
+
+export const changePlayerState = async (
+  command: PlayerCommand,
+  interaction: ChatInputCommandInteraction,
+  shoukaku: Shoukaku,
+) => {
+  const guildMember = interaction.member as GuildMember;
+
+  if (!guildMember?.voice?.channel?.id) {
+    await interaction.reply(`You're not connected to any voice channel! >:(`);
+    return;
+  }
+
+  let player = shoukaku.players.get(interaction.guildId as string);
+
+  switch (command) {
+    case PlayerCommand.PAUSE:
+      player?.setPaused(true);
+      await interaction.reply('Paused');
```

Comment:
I think you should give some feedback to the user if they are trying to resume/pause a track that the player has already dropped

Reply:
Can you elaborate a bit? Wdym by a track that the player has dropped

Reply:
Like if a track has already ended or was stopped?

Reply:
Yeah, when the player ends a track or the user stops it. The user does not get any feedback when trying to resume/stop. 
Same if the player unexpectedly "drops" a track due to some error and user tried to resume/stop.

Reply:
Right. I can handle cases where the song was stopped manually. However, in cases where the bot was forced to disconnect while playing a song, I can't find a reliable method to detect if the bot is connected to a voice channel or not.

I *could* manually kill the player and connection by attaching a listener to the event that triggers when a bot is disconnected, but moving the player around in voice channels also count as disconnects, and I don't want the music to stop when the bot is moved around.